assets/js/src/integrations/wp-consent-api.js

```diff
@@ -0,0 +1,56 @@
+const consentMap = {
+	statistics: [ 'analytics_storage' ],
+	marketing: [ 'ad_storage', 'ad_user_data', 'ad_personalization' ],
+};
+
+export const setCurrentConsentState = ( {
+	tracker_function_name: trackerFunctionName,
+} ) => {
+	// eslint-disable-next-line camelcase -- `wp_has_consent` is defined by the WP Consent API plugin.
+	if ( typeof wp_has_consent === 'function' ) {
+		if ( window.wp_consent_type === undefined ) {
+			window.wp_consent_type = 'optin';
+		}
+
+		const consentState = {};
+
+		for ( const [ category, types ] of Object.entries( consentMap ) ) {
+			// eslint-disable-next-line camelcase, no-undef
+			if ( wp_has_consent( category ) ) {
+				types.forEach( ( type ) => {
+					consentState[ type ] = 'granted';
+				} );
+			}
```

Comment:
:nail_care: We can add an explanation to this `eslint-disable-next-line` as well.

:question: Related to https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/425#discussion_r1608711163

```suggestion
			types.forEach( ( type ) => {
				// eslint-disable-next-line camelcase, no-undef
				consentState[ type ] = wp_has_consent( category )
					? 'granted'
					: 'denied';
			} );
```

AFAIUnderstood the docs `gtag('consent', 'update', ... )` is for the current visitor, so the region does not matter. If they disallow, I assume they mean to disallow tracking their data regardless where they connect from at the moment.

Reply:
Thanks for the explanations, now I get it and agree :)